packages/app/obojobo-document-engine/src/scripts/oboeditor/components/toolbars/file-toolbar-viewer.js

```diff
@@ -46,14 +58,44 @@ const FileToolbarViewer = props => {
 			)
 		})()
 
-		const insertMenuItems = insertableItems.map(item => ({
-			name: item.name,
-			action: () => {
-				Transforms.insertNodes(editor, item.cloneBlankNode())
-				ReactEditor.focus(editor)
-			},
-			disabled: isInsertDisabledForItem(selectedNodes, item.name, sel)
-		}))
+		const insertMenuItems = insertableItems.map(item => {
+			return {
+				name: item.name,
+				action: () => {
+					const path = Editor.path(editor, editor.selection, { edge: 'start' })
+					const prevChildrenCount = editor.children.length
+
+					Transforms.insertNodes(editor, item.cloneBlankNode())
+
+					// Since inserting a node inside a figure caption can sometimes
+					// cause figure to be duplicated after the caption is split, we
+					// need to convert the duplicated figure below to a text node.
+					if (
+						containsFigureNode(selectedNodes) &&
```

Comment:
Unfortunately this is counterproductive to our efforts to isolate the chunks/modules from one another.

I'd prefer we ask the selected nodes if they need this behavior OR just let the node handle this behavior:

For instance, perhaps each node can implement handleInsert - check for nodes that do, and execute it. pass the path and editor and selected nodes.  Then the node itself could handle the special requirements

This will be needed for iFrame, Materia, and Figure, so it makes sense to not just look for Figure nodes.